packages/jellyfish-api-core/__tests__/category/masternode/resignMasternode.test.ts

```diff
@@ -0,0 +1,75 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodeState } from '../../../src/category/masternode'
+import { RpcApiError } from '../../../src'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should resignMasternode', async () => {
+    const ownerAddress = await client.wallet.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(ownerAddress)
+
+    await container.generate(1)
+
+    const hex = await client.masternode.resignMasternode(masternodeId)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const resignedMasternode = Object.values(await client.masternode.listMasternodes()).filter(mn => mn.ownerAuthAddress === ownerAddress)
+    for (const masternode of resignedMasternode) {
+      expect(masternode.state).toStrictEqual(MasternodeState.PRE_RESIGNED)
+      expect(masternode.resignTx).toStrictEqual(hex)
+    }
+  })
+
+  it('should resignMasternode with utxos', async () => {
+    const ownerAddress = await client.wallet.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(ownerAddress)
+    const { txid } = await container.fundAddress(ownerAddress, 10)
+
+    await container.generate(1)
+
+    const utxos = (await client.wallet.listUnspent())
+      .filter((utxo: any) => utxo.txid === txid)
+      .map((utxo: any) => {
+        return {
+          txid: utxo.txid,
+          vout: utxo.vout
+        }
+      })
+
+    const hex = await client.masternode.resignMasternode(masternodeId, utxos)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const resignedMasternode = Object.values(await client.masternode.listMasternodes()).filter(mn => mn.ownerAuthAddress === ownerAddress)
+    for (const masternode of resignedMasternode) {
```

Comment:
You need to assert the length of resignedMasternode before calling for-loop. 
If the length is zero, your test will still pass (Which is incorrect).

Reply:
It is done